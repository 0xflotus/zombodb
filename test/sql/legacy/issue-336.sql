create table scaled_float_test(
  id bigserial primary key,
  scale_3 numeric(10,3)
);

select * from zdb.define_field_mapping('scaled_float_test','scale_3','{
  "type" : "scaled_float",
  "scaling_factor": 1000
}');

CREATE INDEX idx_zdb_scaled_float_test
  ON scaled_float_test
  USING zombodb( (scaled_float_test.*));

insert into scaled_float_test (scale_3) values(0.480);
insert into scaled_float_test (scale_3) values(0.636);
insert into scaled_float_test (scale_3) values(0.637);

select id, scale_3,CTID from scaled_float_test where scaled_float_test ==> 'scale_3 < 0.637' order by id;

drop table scaled_float_test;
