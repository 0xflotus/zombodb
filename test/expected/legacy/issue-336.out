create table scaled_float_test(
  id bigserial primary key,
  scale_3 numeric(10,3)
);
select * from zdb.define_field_mapping('scaled_float_test','scale_3','{
  "type" : "scaled_float",
  "scaling_factor": 1000
}');
 define_field_mapping 
----------------------
 
(1 row)

CREATE INDEX idx_zdb_scaled_float_test
  ON scaled_float_test
  USING zombodb( (scaled_float_test.*));
insert into scaled_float_test (scale_3) values(0.480);
insert into scaled_float_test (scale_3) values(0.636);
insert into scaled_float_test (scale_3) values(0.637);
select id, scale_3,CTID from scaled_float_test where scaled_float_test ==> 'scale_3 < 0.637' order by id;
 id | scale_3 | ctid  
----+---------+-------
  1 |   0.480 | (0,1)
  2 |   0.636 | (0,2)
(2 rows)

drop table scaled_float_test;
